opencga-app/src/main/java/org/opencb/opencga/app/migrations/v3/v3_2_0/TASK_5964/AddVersionToClinicalAnalysisMigration.java

```diff
@@ -0,0 +1,62 @@
+import org.apache.commons.collections4.CollectionUtils;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.opencb.opencga.catalog.db.mongodb.OrganizationMongoDBAdaptorFactory;
+import org.opencb.opencga.catalog.migration.Migration;
+import org.opencb.opencga.catalog.migration.MigrationTool;
+
+import java.util.Collections;
+import java.util.List;
+
+@Migration(id = "add_version_to_clinicalAnalysis", description = "Add version to Clinical Analysis #TASK-5964", version = "3.2.0",
+        language = Migration.MigrationLanguage.JAVA, domain = Migration.MigrationDomain.CATALOG, date = 20240527)
+public class AddVersionToClinicalAnalysisMigration extends MigrationTool {
+
+    @Override
+    protected void run() throws Exception {
+        // Add missing mandatory "versioning" fields to Clinical Analysis
+        logger.info("Adding missing 'versioning' fields to Clinical Analysis...");
+        Bson versionDoesNotExistQuery = Filters.exists("version", false);
+        Bson projection = Projections.include("release", "interpretation", "secondaryInterpretations");
+        migrateCollection(OrganizationMongoDBAdaptorFactory.CLINICAL_ANALYSIS_COLLECTION, versionDoesNotExistQuery, projection,
+                (document, bulk) -> {
+                    int version = 1;
+                    Document interpretation = document.get("interpretation", Document.class);
+                    if (interpretation != null) {
+                        int interpretationVersion = interpretation.get("version", Number.class).intValue();
+                        version = Math.max(version, interpretationVersion + 1);
+                    }
+                    List<Document> secondaryInterpretations = document.getList("secondaryInterpretations", Document.class);
+                    if (CollectionUtils.isNotEmpty(secondaryInterpretations)) {
+                        for (Document secondaryInterpretation : secondaryInterpretations) {
+                            int secondaryInterpretationVersion = secondaryInterpretation.get("version", Number.class).intValue();
+                            version = Math.max(version, secondaryInterpretationVersion + 1);
+                        }
+                    }
+                    int release = document.get("release", Number.class).intValue();
+                    Document updateDocument = new Document()
+                            .append("version", version)
+                            .append("_releaseFromVersion", Collections.singletonList(release))
+                            .append("_lastOfVersion", true)
+                            .append("_lastOfRelease", true);
+                    bulk.add(new UpdateOneModel<>(Filters.eq("_id", document.get("_id")), new Document("$set", updateDocument)));
+                });
+
+        // Recalculate indexes
+        logger.info("Installing new indexes...");
+        catalogManager.installIndexes(organizationId, token);
```

Comment:
This will create multiple new indexes, not just related to this particular migration... shouldn't this be an independent migration?

Reply:
That will only add new indexes which are the ones that I want to create. Those new indexes have been added for this particular feature so I'd say it makes some sense having it there. I wouldn't create an independent migration just for this. If those indexes already exist, it will not do anything.